### ISSUE TITLE
How do I map enums?

### ISSUE THREAD

**Issue by namedtuple** [MicrosoftDocs/PowerShell-Docs#4173]
Hello!

I have a cmdlet .cdxml file which uses enums instead of classes for a few properties.  Could you provide an example of how to map an enum using this schema.mof format?  Documentation on this schema.mof file format is extremely lacking online.

Thank you very much.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4c360a7c-840b-9e52-914c-23ac66d3ecb7
* Version Independent ID: 679c1633-a2c1-688c-40cb-e273f59ec9f1
* Content: [Authoring the MOF schema file for a Management OData web service](https://docs.microsoft.com/en-us/powershell/developer/webservices/authoring-the-mof-schema-file-for-a-management-odata-web-service#feedback)
* Content Source: [developer/webservices/authoring-the-mof-schema-file-for-a-management-odata-web-service.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/developer/webservices/authoring-the-mof-schema-file-for-a-management-odata-web-service.md)
* Product: **powershell**
* Technology: **powershell-developer**
* GitHub Login: @joeyaiello
* Microsoft Alias: **jaiello**

**Comment by sdwheeler** [comment]
The MOF schema specification is owned by the DTMF. See the following document: https://www.dmtf.org/sites/default/files/standards/documents/DSP0221_3.0.0.pdf

**Comment by namedtuple** [comment]
This looks like it will be extremely helpful - thank you very much!

**Comment by namedtuple** [comment]
Is there an equivalent document for the schema.xml format?

**Comment by sdwheeler** [comment]
Try searching the DMTF site. See if this is what you need: https://www.dmtf.org/standards/cim/cim_schema_v282

**Comment by namedtuple** [comment]
It appears that the Microsoft Management OData IIS Extension does NOT support enumerations despite enums being extensively documented in the DMTF spec you linked.  I've even decompiled the Microsoft.Management.Odata.dll and find no code that looks like it handles parsing enums.

I've discovered I can change the type to String or String[] which works for GET requests, but nothing I've tried works for doing a POST request where a custom enum type is required.
